src/gt4py/next/iterator/transforms/infer_domain.py

```diff
@@ -25,10 +25,29 @@
 from gt4py.next.utils import flatten_nested_tuple, tree_map
 
 
-DOMAIN: TypeAlias = domain_utils.SymbolicDomain | None | tuple["DOMAIN", ...]
+class DomainAccessDescriptor(eve.StrEnum):
+    """
+    Descriptor for domains that could not be inferred.
```

Comment:
Add a note that this is only used internally for checking, then replaced by `None` (if I understand correctly).

Reply:
This is not replaced by None, it appears in the `domain` annex attribute, but is not placed in the IR.

Reply:
Do you need this information (the info what kind of "no domain" it is)? I don't really like that this ad-hoc concept leaks into other passes. Especially, since it is not clear how this Descriptor should be extended. If we call it `NoDomainDescriptor` at least it is clear there is no domain info.

Reply:
TODO: write that DomainAccessDescriptor.KNOWN is missing, but skipped since we don't need it for technical reasons.